src/resource.js

```diff
@@ -1,6 +1,58 @@
 'use strict'
 
-export default () => {
+import url from 'url'
+import jts from 'jsontableschema'
+import _ from 'lodash'
 
-}
+export default class Resource {
 
+  constructor(descriptor) {
+    this._descriptor = descriptor
+    this.REMOTE_PROTOCOLS = ['http:', 'https:', 'ftp:', 'ftps:']
+  }
+
+  getDescriptor() {
+    return this._descriptor
+  }
+
+  getName() {
+    return this.getDescriptor()['name']
+  }
+
+  getType() {
+    const resourceField = this._getResourceField()
+
+    if (resourceField === 'data') {
+      return 'inline'
+    }
+
+    const protocol = url.parse(this.getDescriptor()[resourceField])['protocol']
+    if (_.includes(this.REMOTE_PROTOCOLS, protocol)) {
+      return 'remote'
+    } else {
+      return 'local'
+    }
+
+  }
+
+  getSource() {
+    return this.getDescriptor()[this._getResourceField()]
+  }
+
+  getTable() {
```

Comment:
Only way I know to make this function wait for the promise from jts to resolve and then return a value or in case of exception a `null` is to use `await` (https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await) which is proposal and still not in the standard. Am I missing something here or we will take this approach?